### PR TITLE
add links to code comments in tech notes readme

### PR DESCRIPTION
**Pull request by msbutler**
This patch aims to centralize some of the nuggets of wisdom scattered throughout the codebase.

### CODE REVIEW

**Review by irfansharif — APPROVED** [review]

File: docs/tech-notes/README.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ onboarding and capture organizational knowledge.
 Technical notes differ from RFCs in that they describe the state of
 the project as it *currently* is, not the way we wish it to be.
 
+The codebase also contains a gold mine of large code comments that provide a deeper conceptual dive of some aspect of the codebase. Below is a list of links to some of these comments. Note that these link to master, so if a link dies, please update it!
+- Overview of the Keyspace: https://github.com/cockroachdb/cockroach/blob/master/pkg/keys/doc.go
```
[the remark this answers is not in the document]
Comment (irfansharif) [reply docs/tech-notes/README.md:12]: If you're linking to a specific line in the file, it gets stale pretty quickly as lines are added above and below it. I suspect these comment blocks listed here are useful to find keywords to go grep for and then look for it directly in the codebase if wanting an up-to-date view. A few other big blocks:
- https://github.com/cockroachdb/cockroach/blob/90054c6b7b9629aeb904ffd4c0b9161f70e214b7/pkg/clusterversion/cockroach_versions.go#L18
- https://github.com/cockroachdb/cockroach/blob/d676d7e0a6739661c9993ca8dbb6d14feeb6b33d/pkg/util/hlc/doc.go#L12-L304

**Comment by erikgrinaker** [comment]
Here's another one

https://github.com/cockroachdb/cockroach/blob/c097a16427f65e9070991f062716d222ea5903fe/pkg/sql/opt/doc.go#L12

File: docs/tech-notes/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (msbutler) [reply docs/tech-notes/README.md]: Done. Links go to specific SHAs now.

**Comment by msbutler** [comment]
bors r = irfansharif

**Comment by msbutler** [comment]
bors r=irfansharif

**Pull request merged**